### PR TITLE
fix GROWTH_RATE comments bug

### PR DESCRIPTION
**Pull request by toywei**
It also exists in  3.7/3.6/3.3.

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@toywei

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by toywei** [comment]
I had.

**Comment by brandtbucher** [comment]
Thanks for your time @toywei, and welcome to CPython! :sunglasses:

The CLA usually takes a business day or two to process, so I'd check back again tomorrow.

File: Objects/dictobject.c [hunk 1]
```diff
@@ -414,7 +414,7 @@ dictkeys_set_index(PyDictKeysObject *keys, Py_ssize_t i, Py_ssize_t ix)
 
 /* GROWTH_RATE. Growth rate upon hitting maximum load.
  * Currently set to used*3.
- * This means that dicts double in size when growing without deletions,
+ * This means that dicts three times in size when growing without deletions,
```
Comment (brandtbucher) [inline Objects/dictobject.c:417]: If this is correct, though, the wording can be improved slightly:

```suggestion
 * This means that dicts triple in size when growing without deletions,
```

**Review by brandtbucher — CHANGES_REQUESTED** [review]
I'm not a dictionary expert, but the existing wording seems correct. My understanding is that whenever the dict becomes 2/3 full, it resizes to triple the number of *used* slots (not *total* ones).

So effectively, it does indeed double in size.

**Comment by methane** [comment]
@brandtbutcher is right.
Dict size grows from used*3/2 to used*3.

File: Objects/dictobject.c (hunk 1 above)
Comment (tim-one) [reply Objects/dictobject.c:417]: Yes, for a dict with `N` slots, at most about `2*N/3` can be used.  That's the maximum value of `ma_used`.  So tripling `ma_used` then gives a dict with `3*(2*N/3) ` = `2*N` slots, only doubling (not tripling) the total size.

**Comment by brandtbucher** [comment]
Okay then. This can probably be closed?

File: Objects/dictobject.c (hunk 1 above)
Comment (toywei) [reply Objects/dictobject.c:417]: If so ，
https://github.com/python/cpython/blob/3.5/Objects/dictobject.c
#define GROWTH_RATE(d) (((d)->ma_used*2)+((d)->ma_keys->dk_size>>1))
https://github.com/python/cpython/blob/3.6/Objects/dictobject.c
#define GROWTH_RATE(d) (((d)->ma_used*2)+((d)->ma_keys->dk_size>>1))
(d)->ma_keys->dk_size>>1 === (d)->ma_used*1 ？

Comment (tim-one) [reply Objects/dictobject.c:417]: Why are you showing out-of-date code?  The current (master) definition is
```c
#define GROWTH_RATE(d) ((d)->ma_used*3)
```
and that's what everyone else here is talking about.

**Pull request closed without merging**